make python udf serialization format to be binary plus tensor tables

zhaojuanmao opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27191 skip all rpc and dist autograd spawn tests for <PY36
* **#27136 make python udf serialization format to be binary plus tensor tables**

make python udf serialization format to be binary plus tensor tables, so that tensors can be attached to autograd graph, handled in the same way as builtin operators

Differential Revision: [D17405686](https://our.internmc.facebook.com/intern/diff/D17405686/)

zhaojuanmao committed:
make python udf serialization format to be binary plus tensor tables

make python udf serialization format to be binary plus tensor tables, so that tensors can be attached to autograd graph, handled in the same way as builtin operators

Differential Revision: [D17405686](https://our.internmc.facebook.com/intern/diff/D17405686/)

[ghstack-poisoned]

mrshenli reviewed (approved):
This is the dup of #26314. The previous one hit merge conflicts during landing, but the OSS PR was already closed.

zhaojuanmao committed:
Update on "make python udf serialization format to be binary plus tensor tables"


make python udf serialization format to be binary plus tensor tables, so that tensors can be attached to autograd graph, handled in the same way as builtin operators

Differential Revision: [D17405686](https://our.internmc.facebook.com/intern/diff/D17405686/)

[ghstack-poisoned]

zhaojuanmao commented:
@pytorchbot retest this please

zhaojuanmao commented:
failed tests are not relevant

fmassa commented on torch/distributed/internal_rpc_utils.py:
```diff
@@ -4,30 +4,125 @@
 import copyreg
 import io
 import pickle
+import six
```
This adds a dependency on `six` which wasn't present before.

You should use instead https://github.com/pytorch/pytorch/blob/master/torch/_six.py

fmassa reviewed (commented):
This PR breaks torchvision CI because it adds a hard-dependency on `six` which is not translated to the binaries https://circleci.com/gh/pytorch/vision/20778?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

zhaojuanmao commented:
@fmassa sorry that it broke torchvision CI, wish there is a way to catch this kind of issue before landing diffs:(
